flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxProcessor.java

```diff
@@ -175,6 +199,7 @@ public void close() {
     public void drain() throws Exception {
         for (final Mail mail : mailbox.drain()) {
             mail.run();
+            mailboxThroughputMeter.markEvent();
```

Comment:
I would be slightly worried about adding extra virtual calls overhead. I don't think this is a confirmed concern here, but since we could quite easily avoid it, wouldn't it be better to have here a simple `long counter++` here?

Reply:
Well, with a quick look at SumCounter, I actually don't understand why this needs to extend the SimpleCounter ... looks a bit hacky to me. I assume it should rather implement the Counter interface ... not sure though.

Reply:
> Well, with a quick look at SumCounter, I actually don't understand why this needs to extend the SimpleCounter ... looks a bit hacky to me. I assume it should rather implement the Counter interface ... not sure though.

You can try it out and refactor it if you want/think it's worth it.

I was thinking about one of the following ideas:
1. Pass `SimpleCounter`, not `Counter` to the `MailboxProcessor`. 
2. Defining `public static final class MailboxLengthCounter extends SimpleCounter {}` (or sth like that).

However on a second thought, it shouldn't matter. Every enqueued mail has to go through threads synchronisation while being enqueued, so an extra virtual call here or there should be negligible in comparison. Let's keep it as it is.